.github/workflows/3rd-party-check.yml

```diff
@@ -0,0 +1,84 @@
+name: New 3rd Party Dependency Check
+
+on:
+  pull_request:
+    branches:
+      - staging
+    paths:
+      - 'package.json'
+      - 'package-lock.json'
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  check-3rd-party:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Ensures all history is fetched
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      - name: Check for 3rd Party Dependencies
+        id: check_dependencies
+        run: |
+          # Install required tools
+          sudo apt-get install -y jq
+
+          # Fetch the base branch
+          git fetch origin ${{ github.base_ref }}
+          git checkout ${{ github.base_ref }}
+          git checkout -
+
+          # Initialize a flag
+          echo "DEPENDENCY_CHANGE=false" >> $GITHUB_ENV
+
+          # Compare package.json
+          echo "Checking package.json for changes..."
+          if git diff --name-only origin/${{ github.base_ref }}...HEAD | grep 'package.json'; then
+            echo "Changes detected in package.json"
+            echo "DEPENDENCY_CHANGE=true" >> $GITHUB_ENV
+          fi
+
+          # Compare package-lock.json
+          echo "Checking package-lock.json for changes..."
+          if git diff --name-only origin/${{ github.base_ref }}...HEAD | grep 'package-lock.json'; then
+            echo "Changes detected in package-lock.json"
+            echo "DEPENDENCY_CHANGE=true" >> $GITHUB_ENV
+          fi
+
+      - name: Post Comment
+        if: ${{ env.DEPENDENCY_CHANGE == 'true' }}
```

Comment:
Have you noticed this context access warning for this line?

Reply:
If it is not a problem can ignore.

Reply:
I don't think it's a problem, but having a warning is never a good time. I've resolved it, by setting the conditional if statement to use the outputs from the previous step